File: src/CrowdfundingApp.Common.Core/Messages/Users/UpdateUserRequestMessage.cs
﻿
namespace CrowdfundingApp.Common.Core.Messages.Users
{
    public class UpdateUserRequestMessage : MessageBase
    {
        /// <example>nickName</example>
        public string UserName { get; set; }
        /// <example>test@user.com</example>
        public string Email { get; set; }
        /// <example>Jonh</example>
        public string Name { get; set; }
        /// <example>Dorian</example>
        public string Surname { get; set; }
        /// <example>Middle name</example>
        public string MiddleName { get; set; }
        public string Image { get; set; }
        public string CurrentPassword { get; set; }
        public string NewPassword { get; set; }
        public string ConfirmPassword { get; set; }
    }
}

File: src/CrowdfundingApp.Client/src/app/models/requests/users/UpdateUserRequestMessage.ts
export class UpdateUserRequestMessage {
  constructor(
    public userName?: string,
    public email?: string,
    public name?: string,
    public surname?: string,
    public middleName?: string,
    public image?: string,
    public currentPassword?: string,
    public newPassword?: string,
    public confirmPassword?: string
  ) { }
}
